### PR TITLE
Onna amqp backport

### PR DESCRIPTION
**Pull request by davidonna**
Contains these commits from onna/guillotina_amqp:

6b4aaa8a053fd77838d9bf48f7556fb62508d655

ebd73784cfef1b438037ab2c048b5fc06a0616f2

### CODE REVIEW

**Comment by davidonna** [comment]
[https://github.com/atlasense/product/issues/5354](https://github.com/atlasense/product/issues/5354)

File: guillotina_amqp/state.py [hunk 1]
```diff
@@ -372,3 +386,48 @@ def __init__(self, task_id):
     async def is_canceled(self):
         util = get_state_manager()
         return await util.is_canceled(self.task_id)
+
+
+async def update_task_status(state_manager, task_id, status, task=None, ttl=None,
```
Comment (lferran) [inline guillotina_amqp/state.py]: @vangheem I'm thinking whether we should eventually expire all data from redis... at least after a long period of time, like 24h?

**Comment by davidonna** [comment]
Updated.

File: guillotina_amqp/state.py (hunk 1 above)
Comment (vangheem) [reply guillotina_amqp/state.py]: That is probably fine to set a default ttl.

Although, as long as redis is configured with an eviction policy, it should be fine to not.

**Comment by lferran** [comment]
@vangheem could you ship a new release with this PR?

**Comment by vangheem** [comment]
2.2.0 released!

**Pull request merged**